Mypy false positive for union validator

Not sure if this is a duplicate but I couldn't find the problem in the open issues. If I overlooked it, please close, of course.

```python
from attrs import define, field
from attrs.validators import instance_of

LeTypeAlias = int | str


@define
class LeClass:
    x: LeTypeAlias = field(validator=instance_of(LeTypeAlias))
```

Works perfectly but `mypy` complains about:
```
error: No overload variant of "instance_of" matches argument type "UnionType[int, str]"  [call-overload]
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -66,7 +66,7 @@ jobs:
         run: |
           DO_MYPY=1
 
-          if [[ "$V" == "3.8" ]]; then
+          if [[ "$V" == "3.8" || "$V" == "3.9" ]]; then
             DO_MYPY=0
           fi
 

--- a/changelog.d/1385.change.md
+++ b/changelog.d/1385.change.md
@@ -1,0 +1,2 @@
+`attrs.validators.instance_of()`'s type hints now allow for union types.
+For example: `instance_of(str | int)`

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,10 +30,10 @@ dynamic = ["version", "readme"]
 
 [project.optional-dependencies]
 tests-mypy = [
-  'pytest-mypy-plugins; platform_python_implementation == "CPython" and python_version >= "3.9"',
+  'pytest-mypy-plugins; platform_python_implementation == "CPython" and python_version >= "3.10"',
   # Since the mypy error messages keep changing, we have to keep updating this
   # pin.
-  'mypy>=1.11.1; platform_python_implementation == "CPython" and python_version >= "3.9"',
+  'mypy>=1.11.1; platform_python_implementation == "CPython" and python_version >= "3.10"',
 ]
 tests = [
   # For regression test to ensure cloudpickle compat doesn't break.

--- a/src/attr/validators.pyi
+++ b/src/attr/validators.pyi
@@ -24,7 +24,6 @@ _I = TypeVar("_I", bound=Iterable)
 _K = TypeVar("_K")
 _V = TypeVar("_V")
 _M = TypeVar("_M", bound=Mapping)
-_UT = TypeVar("_UT", bound=UnionType)
 
 def set_disabled(run: bool) -> None: ...
 def get_disabled() -> bool: ...
@@ -47,7 +46,7 @@ def instance_of(
 @overload
 def instance_of(type: tuple[type, ...]) -> _ValidatorType[Any]: ...
 @overload
-def instance_of(type: _UT) -> _ValidatorType[_UT]: ...
+def instance_of(type: UnionType) -> _ValidatorType[Any]: ...
 def optional(
     validator: (
         _ValidatorType[_T]

--- a/src/attr/validators.pyi
+++ b/src/attr/validators.pyi
@@ -1,3 +1,4 @@
+from types import UnionType
 from typing import (
     Any,
     AnyStr,
@@ -44,6 +45,8 @@ def instance_of(
 ) -> _ValidatorType[_T1 | _T2 | _T3]: ...
 @overload
 def instance_of(type: tuple[type, ...]) -> _ValidatorType[Any]: ...
+@overload
+def instance_of(type: UnionType) -> _ValidatorType[_T]: ...
 def optional(
     validator: (
         _ValidatorType[_T]

--- a/src/attr/validators.pyi
+++ b/src/attr/validators.pyi
@@ -24,6 +24,7 @@ _I = TypeVar("_I", bound=Iterable)
 _K = TypeVar("_K")
 _V = TypeVar("_V")
 _M = TypeVar("_M", bound=Mapping)
+_UT = TypeVar("_UT", bound=UnionType)
 
 def set_disabled(run: bool) -> None: ...
 def get_disabled() -> bool: ...
@@ -46,7 +47,7 @@ def instance_of(
 @overload
 def instance_of(type: tuple[type, ...]) -> _ValidatorType[Any]: ...
 @overload
-def instance_of(type: UnionType) -> _ValidatorType[_T]: ...
+def instance_of(type: _UT) -> _ValidatorType[_UT]: ...
 def optional(
     validator: (
         _ValidatorType[_T]

--- a/tests/typing_example.py
+++ b/tests/typing_example.py
@@ -228,6 +228,9 @@ class Validated:
     k: int | str | C = attr.ib(
         validator=attrs.validators.instance_of((int, C, str))
     )
+    kk: int | str | C = attr.ib(
+        validator=attrs.validators.instance_of(int | C | str)
+    )
 
     l: Any = attr.ib(
         validator=attr.validators.not_(attr.validators.in_("abc"))

--- a/tox.ini
+++ b/tox.ini
@@ -3,7 +3,7 @@ min_version = 4
 env_list =
     pre-commit,
     py3{8,9,10,11,12,13}-tests,
-    py3{9,10,11,12,13}-mypy,
+    py3{10,11,12,13}-mypy,
     pypy3,
     pyright,
     docs{,-sponsors},

